src/annotator/anchoring/test/types-test.js

```diff
@@ -19,46 +21,335 @@ describe('types', function () {
+          suffix: ', and dedicated to the proposition that',
+        });
+      });
+    });
+
+    describe('#toRange', () => {
+      it('returns a valid DOM Range', () => {
+        $imports.$restore({
+          'dom-anchor-text-quote': {
+            toRange: true,
+          },
+        });
         const quoteAnchor = new TextQuoteAnchor(container, 'Liberty');
         const range = quoteAnchor.toRange();
         assert.instanceOf(range, Range);
         assert.equal(range.toString(), 'Liberty');
       });
 
-      it('throws if the quote is not found', function () {
+      it('throws if the quote is not found', () => {
+        $imports.$restore({
+          'dom-anchor-text-quote': {
+            toRange: true,
+          },
+        });
         const quoteAnchor = new TextQuoteAnchor(
           container,
           'five score and nine years ago'
         );
-        assert.throws(function () {
+        assert.throws(() => {
           quoteAnchor.toRange();
         });
       });
     });
 
-    describe('#toPositionAnchor', function () {
-      it('returns a TextPositionAnchor', function () {
+    describe('#toPositionAnchor', () => {
+      it('returns a TextPositionAnchor', () => {
+        $imports.$restore({
+          'dom-anchor-text-quote': {
+            toTextPosition: true,
+          },
+        });
         const quoteAnchor = new TextQuoteAnchor(container, 'Liberty');
         const pos = quoteAnchor.toPositionAnchor();
         assert.instanceOf(pos, TextPositionAnchor);
       });
 
-      it('throws if the quote is not found', function () {
+      it('throws if the quote is not found', () => {
+        $imports.$restore({
+          'dom-anchor-text-quote': {
+            toTextPosition: true,
+          },
+        });
         const quoteAnchor = new TextQuoteAnchor(
           container,
           'some are more equal than others'
         );
-        assert.throws(function () {
+        assert.throws(() => {
           quoteAnchor.toPositionAnchor();
         });
       });
     });
+
+    describe('integration tests', () => {
+      beforeEach(() => {
+        // restore dom-anchor-text-quote to test third party lib integration
+        $imports.$restore();
+      });
+
+      it('can convert a Range to TextQuoteSelector and back to a Range', () => {
+        const range = document.createRange();
+        range.setStart(container.firstChild, 0);
+        range.setEnd(container.firstChild, 4);
+        const anchor = TextQuoteAnchor.fromRange(container, range);
+        assert.deepEqual(anchor.toSelector(), {
+          type: 'TextQuoteSelector',
+          prefix: '',
+          suffix: ' score and seven years ago our f',
+          exact: 'Four',
+        });
+        const newRange = anchor.toRange();
+        // startContainer of anchor.toRange() points to <div>
+        // rather than the text node "Four score ..."
+        // assert.equal(anchor.toRange(), range);
```

Comment:
Perhaps this is expected, but I was not sure so I left a comment. The two Ranges have different startContainers.